osu.Game/Rulesets/Difficulty/Skills/StrainSkill.cs

```diff
@@ -57,7 +57,25 @@ public sealed override void Process(DifficultyHitObject current)
                 currentSectionEnd += SectionLength;
             }
 
-            currentSectionPeak = Math.Max(StrainValueAt(current), currentSectionPeak);
+            double strain = StrainValueAt(current);
+            currentSectionPeak = Math.Max(strain, currentSectionPeak);
+
+            // Store the strain value for the object
+            ObjectStrains.Add(strain);
+        }
+
+        /// <summary>
+        /// Calculates the number of strains weighted against the top strain.
+        /// The result is scaled by clock rate as it affects the total number of strains.
+        /// </summary>
+        public double CountDifficultStrains()
```

Comment:
we may want to make this virtual in the future to allow ruleset calculators to define their own summation if they want, although it seems the concept works fine enough outside of osu! ruleset

Reply:
I'd agree, we should probably rename any existing difficulty attributes if we're going to rename the method though

Reply:
AFAIK it's not an issue? We had attribute renames before so it should be fine

Reply:
Yeah can't think of a better one than `CountTopWeightedStrains`, so I'll go with it